installer/install.iss

```diff
@@ -1946,6 +1978,8 @@ begin
     // Restore the settings chosen during a previous install
     RdbExperimentalOptions[GP_BuiltinAddI].Checked:=ReplayChoice('Enable Builtin Interactive Add','Auto')='Enabled';
 #endif
+    if OptionsArePreSet then
+        PagesPreSet:=PagesPreSet+IntToStr(PrevPageID)+',';
```

Comment:
While this certainly works, it is an _awful_ lot of repeated code, and even worse: it is all too easy to forget when adding new pages (or to forget to remove the code when removing pages!).

Besides, I could imagine that it does not even do what we need it to do if there are no experimental built-ins, i.e. if that page is not even shown.

And even if it does what we need it to do, it is not obvious from looking at this code change.

So really, I would like to go the other way round: accumulate the IDs of the pages that we want to show. (And now that I think about it, maybe an even better name than `PagesWithPreviouslyUnseenOptions` would be `PagesToShow`.)

Reply:
>Besides, I could imagine that it does not even do what we need it to do if there are no experimental built-ins, i.e. if that page is not even shown.

It does behave correctly in this case, because all it will do is add a (in this case useless) ID to the pages-to-skip list.   Note there is also an outer #ifdef that controls whether the experimental page is to be shown at all, and the code changes are inside that outer #ifdef, meaning that if no experimental page, then this code won't run either.

Reply:
> It does behave correctly in this case, because all it will do is add a (in this case useless) ID to the pages-to-skip list.

Ah, that's right. Still: harder to reason about that I'd bargain for.

> Note there is also an outer #ifdef that controls whether the experimental page is to be shown at all, and the code changes are inside that outer #ifdef, meaning that if no experimental page, then this code won't run either.

Valid point, too. Playing even more into my argument that it is tricky to confirm the correctness of this patch. I'd prefer a patch that is super easy to confirm, if possible.